src/adaPets.js

```diff
@@ -10,30 +10,56 @@ const BASE_URL = 'http://localhost:3000/pets/';
 // Option functions.
 const listPets = () => {
   // Fill out as part of Wave 1.
+  axios.get(BASE_URL)
+  .then((response) => {
+    setResult(response.data);
+  })
+  .catch((error) => {
+    setError("Unable to list pets. ${error.message}.");
+  });
 };
 
 const showDetails = (selectedPetId) => {
   if (!selectedPetId) {
-    setError("You tried to show details for a pet without selecting it!");
-    
-  }
+    setError("You tried to show details for a pet without selecting it. ${error.message}.") 
+  };
 
   // Fill out as part of Wave 2.
-};
+  axios.get(BASE_URL+selectedPetId)
+  .then((response) => {
+    setResult(response.data);
+  })
+  .catch((error) => {
+    setError ("ERROR 404: Failed to select and show details");
```

Comment:
You should report the actual status here in case it isn't 404:

```suggestion
    setError ("ERROR ${error.response.status}: Failed to select and show details");
```